tests/components/ring/test_init.py

```diff
@@ -32,3 +38,161 @@ async def test_setup(hass: HomeAssistant, requests_mock: requests_mock.Mocker) -
         "https://api.ring.com/clients_api/doorbots/987652/health",
         text=load_fixture("doorboot_health_attrs.json", "ring"),
     )
+
+
+async def test_auth_failed_on_setup(
+    hass: HomeAssistant,
+    requests_mock: requests_mock.Mocker,
+    mock_config_entry: MockConfigEntry,
+    caplog,
+) -> None:
+    """Test auth failure on setup entry."""
+    mock_config_entry.add_to_hass(hass)
+    with patch(
+        "ring_doorbell.Ring.update_data",
+        side_effect=AuthenticationError,
+    ):
+        result = await hass.config_entries.async_setup(mock_config_entry.entry_id)
+        await hass.async_block_till_done()
+        assert result is False
+        assert "Access token is no longer valid. Please set up Ring again" in [
+            record.message for record in caplog.records if record.levelname == "ERROR"
+        ]
+
+        assert DOMAIN not in hass.data
+        assert mock_config_entry.state is ConfigEntryState.SETUP_ERROR
+
+
+async def test_auth_failure_on_global_update(
+    hass: HomeAssistant,
+    requests_mock: requests_mock.Mocker,
+    mock_config_entry: MockConfigEntry,
+    caplog,
+) -> None:
+    """Test authentication failure on global data update."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    with patch(
+        "ring_doorbell.Ring.update_devices",
+        side_effect=AuthenticationError,
+    ):
+        await hass.data[DOMAIN][mock_config_entry.entry_id][
+            "device_data"
+        ].async_refresh_all()
```

Comment:
We typically do not poke inside of test internals like this. If you want to have a test for this, it needs to exercise the integration APIs (e.g. service call, etc).